### ISSUE TITLE
Only compare if filenames are identical

### ISSUE THREAD

**Issue by dingens** [adrianlopezroche/fdupes#195]
(Having seen your opinion on #58, but I think this is a little different so I ask anyway. Feel free to decline, of course!)

I sometimes have the use case that _if files are duplicates, then they also have the same name_. In that case, the comparison phase could be sped up a lot because no hashing is needed.

I suggest an option `--same-filename` or so that causes fdupes to only do a detailed comparison (hashing, byte-wise comparison) if filenames are equal.

Going further, one could also think of an option `--same-fullpath`: only check files if their full paths (relative to the directory given on command line) are equal.

**Comment by drws** [comment]
I agree with @Grinzold, it'd be a very useful feature. Also, time savings could be huge in certain scenarios (depending on the structure of input data and what one wants to achieve).
